the key for clientAuthorizations.add must be the same as the one of the SecurityDefinition

fretje opened the pull request:
see securitydefinition defined in startup.cs 

https://github.com/aspnetboilerplate/module-zero-core-template/blob/8eb2944430aa814557112c21769c70231f69b911/aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Host/Startup/Startup.cs#L72

see also https://github.com/swagger-api/swagger-ui/issues/1244#issuecomment-245638234

acjh commented:
No, this won't work. The code currently uses `ApiKeyAuthorization` to mimic JWT authentication, so the header must be `abp.auth.tokenHeaderName` (please find out what that is before replacing it with a magic string). If you want to "fix" this, find out if bearer authentication is supported and configure that.

fretje commented:
Well, the header is actually still abp.auth.tokenHeaderName (see the line before my edit). 

What I changed is the key for the add operation of clientAuthorizations, which must be identical to the key set in the securitydefinition for the swagger document. That key is now in startup.cs also a "magic string".

To test this, I added a simple call to abp.swagger.login() at the end of on-complete.js. Then some dialogs appear which ask for tenant, username and password, and then you don't have to manually log in.

I can only say that it wasn't working for me until I made the change in this pull request.

acjh commented:
Which "line before my edit"?

`abp.swagger.login()` is a helper with an independent mechanism: https://github.com/aspnetboilerplate/aspnetboilerplate/issues/2454#issuecomment-326465338

fretje commented on aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Host/wwwroot/swagger/ui/on-complete.js:
```diff
@@ -11,7 +11,7 @@ var abp = abp || {};
             return false;
         }
         var cookieAuth = new SwaggerClient.ApiKeyAuthorization(abp.auth.tokenHeaderName, 'Bearer ' + authToken, 'header');
```
@acjh I'm talking about this line. Here the headername gets set.

acjh commented:
The `name` matches:

https://github.com/aspnetboilerplate/module-zero-core-template/blob/9e63be58c6fd5ab5ac788e10e05d677b5ca715d0/aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Host/Startup/Startup.cs#L75

fretje commented:
The name of the header, yes indeed.

But the name of what you add to swaggerUi.api.clientAuthorizations should match

https://github.com/aspnetboilerplate/module-zero-core-template/blob/9e63be58c6fd5ab5ac788e10e05d677b5ca715d0/aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Host/Startup/Startup.cs#L72

acjh commented:
Did you try `abp.swagger.login()` without the change?

fretje commented:
Yes, I did.
On the master branch of this repo. When I run abp.swagger.login(), I get this in the console:
![image](https://user-images.githubusercontent.com/775707/31945141-95bf4a72-b8ce-11e7-9fd5-4589c0cd8339.png)

But when I try a request after that with swagger ui, I still get 401 (Unauthorized):
![image](https://user-images.githubusercontent.com/775707/31944812-9c773fa6-b8cd-11e7-84bd-c3cbd31349f0.png)
When I make the change in this pull request, all goes well.

acjh commented:
I see. Thank you for the clarification.

hikalkan commented:
So, should I merge this :)

acjh commented:
Yes 👍

acjh commented:
Explanation: The original PRs were tested separately and the helper works on its own, but adding the security definition requires that the client authorization added by the helper to have the same name.
